### DISCUSSION TITLE
Problem running interactive-auth example on native Win11 with msys2

### DISCUSSION THREAD

**Discussion by andyredhead** [sreeise/graph-rs-sdk#492] [Q&A]
Hi,

I'm relatively new to Rust (especially Rust on Win11) and not overly familiar with MS Graph API, so this might be a silly question...

How should I run the "interactive auth" example?

I'm using a Win11 Intel based laptop with Rust 1.80.1 (installed using rustup-init.exe) and the msys2 toolchain, aiming at using a "native" windows platform (rather than WSL2).

I've cloned the git repo for this project.

Running cargo build from a dos console runs ok :)

I'm assuming that to run the interactive auth example I should do:

`graph-rs-sdk>cargo run --features="interactive-auth" --example interactive_auth`

When I run this, cargo reports:

```
   Compiling graph-rs-sdk v2.0.1 (C:\ar-work\project\tinkering\rust-ms-graph-api\graph-rs-sdk)
warning: `graph-oauth` (lib) generated 1 warning (run `cargo fix --lib -p graph-oauth` to apply 1 suggestion)
   Compiling test-tools v0.0.1 (C:\ar-work\project\tinkering\rust-ms-graph-api\graph-rs-sdk\test-tools)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1m 29s
     Running `target\debug\examples\interactive_auth.exe`
```
but nothing seems to happen...

Running `target\debug\examples\interactive_auth.exe` directly from the command line shows a dialog box with an error message:

"The code execution cannot proceed because WebView2Loader.dll was not found. Reinstalling the program may fix this problem."

Looks like WebView2Loader.dll has been built in:

```
target\debug\build\webview2-com-sys-1773cef3f2bbc7a3\out\arm64
target\debug\build\webview2-com-sys-1773cef3f2bbc7a3\out\x64
target\debug\build\webview2-com-sys-1773cef3f2bbc7a3\out\x86
```

I tried copying the x64 dll into target\debug\examples (which is where interactive_auth.exe got built) but that just made the program fail silently.

Any suggestions on how to get this working would be most welcome.

Thanks, Andy

**Comment by sreeise** [comment]
> Hi,
> 
> I'm relatively new to Rust (especially Rust on Win11) and not overly familiar with MS Graph API, so this might be a silly question...
> 
> How should I run the "interactive auth" example?
> 
> I'm using a Win11 Intel based laptop with Rust 1.80.1 (installed using rustup-init.exe) and the msys2 toolchain, aiming at using a "native" windows platform (rather than WSL2).
> 
> I've cloned the git repo for this project.
> 
> Running cargo build from a dos console runs ok :)
> 
> I'm assuming that to run the interactive auth example I should do:
> 
> `graph-rs-sdk>cargo run --features="interactive-auth" --example interactive_auth`
> 
> When I run this, cargo reports:
> 
> ```
>    Compiling graph-rs-sdk v2.0.1 (C:\ar-work\project\tinkering\rust-ms-graph-api\graph-rs-sdk)
> warning: `graph-oauth` (lib) generated 1 warning (run `cargo fix --lib -p graph-oauth` to apply 1 suggestion)
>    Compiling test-tools v0.0.1 (C:\ar-work\project\tinkering\rust-ms-graph-api\graph-rs-sdk\test-tools)
>     Finished `dev` profile [unoptimized + debuginfo] target(s) in 1m 29s
>      Running `target\debug\examples\interactive_auth.exe`
> ```
> 
> but nothing seems to happen...
> 
> Running `target\debug\examples\interactive_auth.exe` directly from the command line shows a dialog box with an error message:
> 
> "The code execution cannot proceed because WebView2Loader.dll was not found. Reinstalling the program may fix this problem."
> 
> Looks like WebView2Loader.dll has been built in:
> 
> ```
> target\debug\build\webview2-com-sys-1773cef3f2bbc7a3\out\arm64
> target\debug\build\webview2-com-sys-1773cef3f2bbc7a3\out\x64
> target\debug\build\webview2-com-sys-1773cef3f2bbc7a3\out\x86
> ```
> 
> I tried copying the x64 dll into target\debug\examples (which is where interactive_auth.exe got built) but that just made the program fail silently.
> 
> Any suggestions on how to get this working would be most welcome.
> 
> Thanks, Andy

Hey Andy, this is a weird one. Just running regular Windows 11 should work with no issue. I am not sure whether its an issue with your webview or something else. 

I tried looking up that error and there may be multple reasons why you would get that from what I gather. 

But one thing im concerned about is that the tao/wry dependencies could need updating due to a bug. So, could you possibly try first doing a `cargo update` then try running it again. If that works then great.

If not, since you have cloned the repo locally you can see if the latest versions of tao and wry are being used here:

https://github.com/sreeise/graph-rs-sdk/blob/804467c7d34f38c850e6174ee3d5d3485ddebdaa/graph-oauth/Cargo.toml#L35

and

https://github.com/sreeise/graph-rs-sdk/blob/804467c7d34f38c850e6174ee3d5d3485ddebdaa/graph-oauth/Cargo.toml#L38

Update those to the latest version and try to see if that works. They tend to update those more than I publish a new version of this crate so they may have put a fix in.

**Reply by andyredhead** [reply]
Hi, thanks for getting back to me.

Updated Cargo with new versions:
```
tao = { version = "0.30.0", features = ["serde"], optional = true }
wry = { version = "0.43.1", optional = true }
```
Getting the code to compile with this update required a minor change in 3 files:
```
                // Disables file drop
                //.with_file_drop_handler(|_| true)
                .with_drag_drop_handler(|_| true)
```
(looks like with_file_drop_handler got renamed to with_drag_drop_handler)

Unfortunately this doesn't seem to have made any difference, I still get the same error message.

For now I will crack on with using an AzureAD AppReg and EnvironmentCredential::client_secret_credential() in a linux cli (which is working ok).

Cheers, Andy

**Reply by sreeise** [reply]
Another thing you can try is to repair the webview program:

Start Menu -> Add or Remove Programs

In the search bar type Webview and you should see a program that says something like "Microsoft Edge Webview2 Runtime".

Click on the elipsis or three dots and then click modify. When the popup comes up you should see a button to start a repair.
